lumigator/python/mzai/backend/backend/tests/integration/api/routes/test_api_workflows.py

```diff
@@ -45,20 +42,15 @@ def test_upload_data_launch_job(local_client: TestClient, dialog_dataset):
         "config_template": "string",
     }
 
-    # Mocking the background task because the test hangs indefinitely,
-    # waiting for the background task to finish.
-    with patch("fastapi.BackgroundTasks.add_task") as mock:
-        create_evaluation_job_response = local_client.post(
-            "/jobs/evaluate/", headers=headers, json=payload
-        )
-        assert create_evaluation_job_response.status_code == 201
-        assert mock.called
+    create_evaluation_job_response = local_client.post(
```

Comment:
Not sure I'm understanding this integration test: it runs an evaluation before inference?
Wouldn't we want a test where the evaluation takes the output of that inference? Not necessarily the quality of the predictions (I agree that wouldn't be the point), but that the inference job inserted the new dataset and the evaluation job can successfully retrieve it.